examples/swapi/src/PersonById.re

```diff
@@ -0,0 +1,48 @@
+let ste = ReasonReact.string;
+
+type action =
+  | ChangeId(string);
+
+type state = {id: string};
+
+let handleClick = event =>
+  if (! ReactEventRe.Mouse.defaultPrevented(event)) {
+    ReactEventRe.Mouse.preventDefault(event);
+  };
+
+let reducer = (action, state) =>
+  switch (action) {
+  | ChangeId(id) => ReasonReact.Update({...state, id})
```

Comment:
Nitpick: since you only have `id` in state, there is no point in spreading the previous `state`

Reply:
it's actually better in case you add other fields in state

Reply:
@emmenko i left it cuz what @Gregoirevda said, and also because when i took it out, code stopped compiling. What is the syntax when you pull out the spread?

Reply:
Without spreading is just `ReasonReact.Update({id})`.

Anyway, it's just a nitpick and you probably get a warning. It's up to you to pick your preferred option 😇

Reply:
If you change to id it reformats to below. Note no brackets around id:
![screen shot 2018-05-24 at 10 57 27 am](https://user-images.githubusercontent.com/2370391/40475206-4ca9acb4-5f41-11e8-8973-23c88b690a8b.png)
```re
let reducer = (action, state) =>
  switch (action) {
  | ChangeId(id) => ReasonReact.Update(id)
  };
```
and you get this error:

```
  This has type:
    (action, state) => ReasonReact.update(string, 'a, 'b)
  But somewhere wanted:
    (action, state) => ReasonReact.update(state, 'c, action)

  The incompatible parts:
    string
    vs
    state

ninja: build stopped: subcommand failed.
>>>> Finish compiling(exit: 1)
```
So what is the correct way to rewrite it?

Reply:
Weird, I thought it would be pawned.

Anyway, using `{ id: id }` works.

<img width="589" alt="image" src="https://user-images.githubusercontent.com/1110551/40477925-fc1e6eae-5f47-11e8-9336-6fedad9a090d.png">

Reply:
Proof for the warning

<img width="844" alt="image" src="https://user-images.githubusercontent.com/1110551/40477977-1da8053a-5f48-11e8-995c-81eaaec1ca08.png">

Reply:
But again, it's just a nitpick 😇